server/src/main/java/org/opensearch/action/support/replication/ReplicationProxy.java

```diff
@@ -18,27 +20,55 @@
  *
  * @opensearch.internal
  */
-public abstract class ReplicationProxy<ReplicaRequest> {
+public abstract class ReplicationProxy<ReplicaRequest extends ReplicationRequest<ReplicaRequest>> {
+
+    /**
+     * This is the replicas proxy which is used for full replication.
+     */
+    private final ReplicationOperation.Replicas<ReplicaRequest> replicasProxy;
+
+    public ReplicationProxy(ReplicationOperation.Replicas<ReplicaRequest> replicasProxy) {
+        this.replicasProxy = replicasProxy;
+    }
 
     /**
      * Depending on the actual implementation and the passed {@link ReplicationMode}, the replication
      * mode is determined using which the replication request is performed on the replica or not.
      *
      * @param proxyRequest                     replication proxy request
-     * @param originalPerformOnReplicaConsumer original performOnReplica method passed as consumer
+     * @param performOnReplicasProxyBiConsumer performOnReplicasProxy
      */
-    public void performOnReplicaProxy(
+    final void performOnReplicaProxy(
         ReplicationProxyRequest<ReplicaRequest> proxyRequest,
-        Consumer<ReplicationProxyRequest<ReplicaRequest>> originalPerformOnReplicaConsumer
+        BiConsumer<
+            Consumer<ActionListener<ReplicationOperation.ReplicaResponse>>,
+            ReplicationProxyRequest<ReplicaRequest>> performOnReplicasProxyBiConsumer
     ) {
         ReplicationMode replicationMode = determineReplicationMode(proxyRequest.getShardRouting(), proxyRequest.getPrimaryRouting());
         // If the replication modes are 1. Logical replication or 2. Primary term validation, we let the call get performed on the
         // replica shard.
-        if (replicationMode == ReplicationMode.FULL_REPLICATION || replicationMode == ReplicationMode.PRIMARY_TERM_VALIDATION) {
-            originalPerformOnReplicaConsumer.accept(proxyRequest);
+        if (replicationMode == ReplicationMode.NO_REPLICATION) {
+            return;
         }
+        performOnReplicaProxy(proxyRequest, replicationMode, performOnReplicasProxyBiConsumer);
     }
 
+    /**
+     * The implementor can decide the {@code Consumer<ActionListener<ReplicationOperation.ReplicaResponse>>} basis the
+     * proxyRequest and replicationMode. This will ultimately make the calls to replica.
+     *
+     * @param proxyRequest                     replication proxy request
+     * @param replicationMode                  replication mode
+     * @param performOnReplicasProxyBiConsumer performOnReplicasProxy
+     */
+    protected abstract void performOnReplicaProxy(
+        ReplicationProxyRequest<ReplicaRequest> proxyRequest,
+        ReplicationMode replicationMode,
+        BiConsumer<
```

Comment:
nit: fix line breaks?

Reply:
this is done by spotless apply. Otherwise spotless check fails.

Reply:
However, this is fixed now due to reducing the variable name length.